lib/sparql.jison

```diff
@@ -266,6 +266,11 @@
     }
     return merged;
   }
+
+  // Return the id of an expression
+  function getIdOfExpression(expression) {
+    return expression.variable ? expression.variable.id : expression.id || expression.expression.id;
```

Comment:
Just wondering if that last case is always sufficient; or do we want:

```suggestion
    return expression.variable ? expression.variable.id : expression.id || getIdOfExpression(expression.id);
```

Also, the nesting of `? : ||` might be confusing; parentheses could help.

Reply:
We actually want the id of the outer expression. The first case is that it is indeed an expression with an 'AS' (explicit like: ?O1 + ?O2) AS ?O12 or implicit like: count(?O) where a free variable name is added to the expression). The second case is for a variable, which directly has an id field, but can also be packed as an expression, so then it is in the expression field and then we are sure there is just a variable with an id field.

Reply:
Sounds good, perhaps add a quick comment with this?